tests/test_runlib.py

```diff
@@ -188,6 +189,108 @@ def test_base_path_is_parent_dir(self):
+      record_artifacts_as_dict(["."], lstrip_paths=lstrip_paths)
+    os.remove(path)
+    os.rmdir("subdir_new")
+
+
+  def test_lstrip_paths_invalid_prefix_directory(self):
+    lstrip_paths = ["not/a/directory/"]
+    expected_artifacts = sorted(["bar", "foo", "subdir/foosub1",
+                                 "subdir/foosub2", "subdir/subsubdir/foosubsub"])
+    artifacts_dict = record_artifacts_as_dict(["."],
+        lstrip_paths=lstrip_paths)
+    self.assertListEqual(sorted(list(artifacts_dict.keys())),
+                         expected_artifacts)
+
+
+  def test_lstrip_paths_valid_prefix_file(self):
+    lstrip_paths = ["subdir/subsubdir/"]
+    expected_artifacts = sorted(["foosubsub"])
+    artifacts_dict = record_artifacts_as_dict(["./subdir/subsubdir/foosubsub"],
+        lstrip_paths=lstrip_paths)
+    self.assertListEqual(sorted(list(artifacts_dict.keys())),
+        expected_artifacts)
+
+
+  def test_lstrip_paths_non_unique_key_file(self):
+    os.mkdir("subdir/subsubdir_new")
+    path = "subdir/subsubdir_new/foosubsub"
+    shutil.copy("subdir/subsubdir/foosubsub", path)
+    lstrip_paths = ["subdir/subsubdir/", "subdir/subsubdir_new/"]
+    with self.assertRaises(in_toto.exceptions.PrefixError):
+      record_artifacts_as_dict(["subdir/subsubdir/foosubsub",
+          "subdir/subsubdir_new/foosubsub"], lstrip_paths=lstrip_paths)
+    os.remove(path)
+    os.rmdir("subdir/subsubdir_new")
+
+
+  @unittest.skipIf(os.getenv("TEST_SKIP_WINDOWS_PY3"), "unicode paths tested separately for windows and py3")
+  def test_lstrip_paths_valid_unicode_prefix_file(self):
+    # Try to create a file with unicode character
+    try:
+      os.mkdir("ಠ")
+      path = "ಠ/foo"
+      shutil.copy("foo", path)
+
+      # Attempt to left strip the path now that the file has been created
+      lstrip_paths = ["ಠ/"]
+      expected_artifacts = sorted(["foo"])
+      artifacts_dict = record_artifacts_as_dict(["./ಠ/"],
+          lstrip_paths=lstrip_paths)
+      self.assertListEqual(sorted(list(artifacts_dict.keys())),
+          expected_artifacts)
+      os.remove(path)
+      os.rmdir("ಠ")
+    except OSError:
+      # OS doesn't support unicode explicit files
+      pass
+
+  @unittest.skipUnless(os.getenv("TEST_SKIP_WINDOWS_PY3"), "test only applies to windows and py3")
+  def test_lstrip_paths_valid_unicode_prefix_file_windows_py_3(self):
+    # Try to create a file with unicode character
+    try:
+      os.mkdir("ಠ")
+      path = "ಠ/foo"
+      shutil.copy("foo", path)
+
+      # Attempt to left strip the path now that the file has been created
+      lstrip_paths = ["ಠ/"]
+      expected_artifacts = sorted(["foo"])
+      artifacts_dict = record_artifacts_as_dict(["./ಠ/"],
+          lstrip_paths=lstrip_paths)
+      self.assertListEqual(sorted(list(artifacts_dict.keys())),
+          expected_artifacts)
+      os.remove(path)
+      os.rmdir("ಠ")
+    except OSError:
+      # OS doesn't support unicode explicit files
+      pass
```

Comment:
Why you distinguish between environments and then run the exact same test in both environments?